### ISSUE TITLE
Update README with demo showing autovectorization for batching

### ISSUE THREAD

**Issue by jessebett** [FluxML/Hydra.jl#12]
If Hydra is offering something similar to Jax's `vmap` then the README sholuld contain a demonstration of usage analogous to [the vmap section in Jax documentation](https://github.com/google/jax#auto-vectorization-with-vmap) or [section 5 of this blog post](https://colinraffel.com/blog/you-don-t-know-jax.html). In particular, the demo should show the common use case in ML, which is to write the objective as though it was elementwise and use Hydra to automatically vectorize the code over a minibatch of elements. It needs to show composability with gradients, e.g. by averaging the elementwise loss gradients over the minibatch. Also it should show how the performance/generated code compares to the equivalent loss function with user-written manually-batched code.
